kubectl-plugin/kubectl-waiops

```diff
@@ -16,9 +42,228 @@ fi
+if [[ "$1" == "upgrade-status" ]]
+then
+    # Script to check upgrade status of CP4WAIOps install for v3.3
+    # This script checks to see if your Cloud Pak for Watson AIOps
+    # instance is properly configured post-upgrade for v3.3. 
+
+    FAILING_UPGRADE=""
+    SUCCESSFULLY_UPGRADED=""
+
+    aiopsEdgeBaseUpgradeStatus() {    
+        UPGRADED=$(oc get aiopsedge aiopsedge -o jsonpath='{.status.conditions[?(@.type=="UpgradeReady")].status}')
+        DETAILS=$(oc get aiopsedge -A  -o custom-columns="KIND:kind,NAMESPACE:metadata.namespace,NAME:metadata.name,STATUS:status.phase")
+        if [ "${UPGRADED}" == "True" ];
+        then 
+            SUCCESSFULLY_UPGRADED+="${newline}${DETAILS}"
+            return 0;
+        else 
+            FAILING_UPGRADE+="${newline}${DETAILS}"
+            return 1;
+        fi
+    }
+
+    aiopsUIUpgradeStatus() {    
+        CURRENT_MAJOR_VERSION=$(oc get aiopsui aiopsui-instance -o jsonpath='{.status.versions.reconciled}' | cut -c1-3)
+        DETAILS=$(oc get aiopsui -A  -o custom-columns="KIND:kind,NAMESPACE:metadata.namespace,NAME:metadata.name,VERSION:status.versions.reconciled")
+        if [ "${CURRENT_MAJOR_VERSION}" == "3.3" ];
+        then 
+            SUCCESSFULLY_UPGRADED+="${newline}${DETAILS}"
+            return 0;
+        else 
+            FAILING_UPGRADE+="${newline}${DETAILS}"
+            return 1;
+        fi
+    }
+
+    kongUpgradeStatus() {    
+        INITIALIZED=$(oc get kong gateway -o jsonpath='{.status.conditions[?(@.type=="Initialized")].status}')
+        DEPLOYED=$(oc get kong gateway -o jsonpath='{.status.conditions[?(@.type=="Deployed")].status}')
+        DETAILS=$(oc get kong -A -o custom-columns="KIND:kind,NAMESPACE:metadata.namespace,NAME:metadata.name")
+        if [ "${INITIALIZED}" == "True" ] && [ "${DEPLOYED}" == "True" ];
+        then 
+            SUCCESSFULLY_UPGRADED+="${newline}${DETAILS}"
+            return 0;
+        else 
+            FAILING_UPGRADE+="${newline}${DETAILS}"
+            return 1;
+        fi
+    }
+
+    aiManagerUpgradeStatus() {    
+        PHASE_STATUS=$(oc get aimanager aimanager -o jsonpath='{.status.phase}')
+        CURRENT_MAJOR_VERSION=$(oc get aimanager aimanager -o jsonpath='{.status.versions.reconciled}' | cut -c1-3)
+        DETAILS=$(oc get aimanager -A -o custom-columns="KIND:kind,NAMESPACE:metadata.namespace,NAME:metadata.name,VERSION:status.versions.reconciled,STATUS:status.phase")
+        if [ "${PHASE_STATUS}" == "Completed" ] && [ "${CURRENT_MAJOR_VERSION}" == "2.4" ];
+        then 
+            SUCCESSFULLY_UPGRADED+="${newline}${DETAILS}"
+            return 0;
+        else 
+            FAILING_UPGRADE+="${newline}${DETAILS}"
+            return 1;
+        fi
+    }
+
+    irCoreUpgradeStatus() {
+        UPGRADED=$(oc get ircore aiops -o json | jq '.metadata.generation as $generation|[(.status.conditions[]|select(.type=="Ready")|.status == "True" and .observedGeneration == $generation)]|all')
```

Comment:
We are now adding a requriement for jq. I'm not sure we want to impose that limitation.

Maybe we can _add_ more information if we have jq in the path, otherwise we should still function